src/main/java/org/testng/internal/reflect/ReflectionHelper.java

```diff
@@ -38,6 +38,36 @@
     return result;
   }
 
+  /**
+   * @return An array of all locally declared methods or equivalent thereof
+   * (such as default methods on Java 8 based interfaces that the given class
+   * implements) but excludes the <code>main()</code> method alone.
+   */
+  public static Method[] excludingMain(Class<?> clazz) {
+    Method[] declaredMethods = clazz.getDeclaredMethods();
+    List<Method> pruned = new LinkedList<>();
+    for (Method declaredMethod :declaredMethods) {
+      if ("main".equals(declaredMethod.getName()) && isStaticVoid(declaredMethod) && acceptsStringArray(declaredMethod)) {
+        continue;
+      }
+      pruned.add(declaredMethod);
+    }
+    return pruned.toArray(new Method[0]);
```

Comment:
I didn't realize that the implementation of `toArray()` in `LinkedList` involved reflection. I thought it would be the same that is being used by `ArrayList`